### PR TITLE
Support scan pandas dict

### PR DESCRIPTION
**Pull request by acquamarin**
Implements scanning from pandas dictionary.

### CODE REVIEW

File: tools/python_api/test/test_scan_pandas.py [hunk 1]
```diff
@@ -599,3 +600,63 @@ def test_scan_empty_list(tmp_path: Path) -> None:
...
     assert tp[0] == "1"
     assert tp[1] == []
+
+
+def test_scan_py_dict_struct_format(tmp_path: Path) -> None:
+    db = kuzu.Database(tmp_path)
+    conn = kuzu.Connection(db)
+    df = pd.DataFrame({
+        "id": [1, 3, 4],
+        "dt": [{'key1': 5, 'key3': 4}, {'key1': 10, 'key3': 25}, None]
```
Comment (andyfengHKU) [inline tools/python_api/test/test_scan_pandas.py:610]: I would also test `key1:None` and empty list, etc...

**Review by andyfengHKU — APPROVED** [review]

File: tools/python_api/test/test_scan_pandas.py (hunk 1 above)
Comment (acquamarin) [reply tools/python_api/test/test_scan_pandas.py:610]: opened #5373
